python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -148,23 +148,106 @@ def construct_log_string(
     )
 
 
+def get_dagster_meta_dict(
+    logging_metadata: DagsterLoggingMetadata, dagster_message_props: DagsterMessageProps
+) -> Dict[str, Any]:
+    # combine all dagster meta information into a single dictionary
+    meta_dict = {
+        **logging_metadata._asdict(),
+        **dagster_message_props._asdict(),
+    }
+    # step-level events can be logged from a pipeline context. for these cases, pull the step
+    # key from the underlying DagsterEvent
+    if meta_dict["step_key"] is None:
+        meta_dict["step_key"] = dagster_message_props.step_key
+
+    return meta_dict
+
+
+class DagsterLogHandler(logging.Handler):
+    def __init__(
+        self,
+        logging_metadata: DagsterLoggingMetadata,
+        loggers: List[logging.Logger],
```

Comment:
If I understand, correctly, these are the loggers that are produced by the Dagster LoggerDefinitions?  I.e. they're kind of handlers in spirit?  If so, I think it's worth calling that out in class docstring.

Reply:
Yep, the only thing in the loggers list will be @loggers. I'll add a docstring explaining this